quiche/src/lib.rs

```diff
@@ -8196,8 +8222,13 @@ impl TransportParams {
                     tp.max_datagram_frame_size = Some(val.get_varint()?);
                 },
 
-                // Ignore unknown parameters.
-                _ => (),
+                // Track unknown transport parameters specially.
+                unknown_tp_id => {
+                    tp.unknown_params.push(UnknownTransportParameter {
```

Comment:
we probably want to put an upper bound on the number of these that we would store, because I don't think there's limit in TLS (other than the full length of the QUIC crypto stream).

I don't know what the bounds should be right now, but as inspiration we have something alongthe same lines for SETTINGS - https://github.com/cloudflare/quiche/blob/9ea2152571bbe48f0a819ebb545f62dcf7d1f738/quiche/src/h3/frame.rs#L50. I don't think it's 100% the same but might be a useful reference point.

Reply:
Let me know if you like the latest revision. I decided against using `smallvec` because I don't think that users will see unknown parameters often enough to warrant pre-allocating space for any entries.